api/src/shared/common/iter_utils.py

```diff
@@ -0,0 +1,13 @@
+from itertools import islice
```

Comment:
Initially wanted to put this file under api/src/utils but ran into import issues when running the tests so I chose the path of least resistance. Happy to move it.